rust/routee-compass-core/src/algorithm/search/search_algorithm_type.rs

```diff
@@ -27,17 +27,14 @@ impl TryFrom<&serde_json::Value> for SearchAlgorithmType {
     /// and returns the SearchAlgorithmType, expected to be a string at the
     /// key "type" on this object.
     fn try_from(config: &serde_json::Value) -> Result<Self, Self::Error> {
-        let type_obj = config
```

Comment:
thanks for catching these, i need to get into the habit of thinking that ok_or is eager and ok_or_else is lazy!